File: EggMoneyna/src/main/java/shinhan/EggMoneyna/monster/dto/MonsterSaveResponseDto.java
package shinhan.EggMoneyna.monster.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import shinhan.EggMoneyna.monster.entity.Monster;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MonsterSaveResponseDto {

    private String name;
    private String feel;
    private String benefit;

    public static MonsterSaveResponseDto of(Monster monster) {
        return new MonsterSaveResponseDto(
                monster.getName().toString(),
                monster.getFeel().toString(),
                monster.getBenefit().toString());
    }
}

File: android/app/src/main/java/com/shbhack/eggmoneyna/data/model/MonsterSaveResponseDto.kt
package com.shbhack.eggmoneyna.data.model

data class MonsterSaveResponseDto(
    val success: String,
    val message: String,
    val data: MonsterSaveDto
)

data class MonsterSaveDto(
    val name: String,
    val feel: String,
    val benefit: String
)
